### PR TITLE
Start of 'flow' app

### PR DESCRIPTION
**Pull request by d3zd3z**
This "flow" app is intended to demonstrate the entire flow of cryptographic information through this process.  It is implemented in Python, using the basic cryptography library.

This first implementation contains a command to generate the initial CA. The Python cryptography library has the advantage of making everything explicit.

Signed-off-by: David Brown <david.brown@linaro.org>

### CODE REVIEW

**Comment by d3zd3z** [comment]
Recreating this from where it was.

File: tools/flow/keys.py [hunk 1]
```diff
@@ -130,7 +130,8 @@ def gen(self, name, ca):
         myname = x509.Name([
             x509.NameAttribute(NameOID.COMMON_NAME, name),
             x509.NameAttribute(NameOID.ORGANIZATION_NAME, u'Linaro Ltd'),
-            x509.NameAttribute(NameOID.ORGANIZATIONAL_UNIT_NAME, u'LinaroCA Device Cert - Signing'),
+            x509.NameAttribute(NameOID.ORGANIZATIONAL_UNIT_NAME,
+                               u'LinaroCA Device Cert - ' + purpose),
```
Comment (microbuilder) [inline tools/flow/keys.py]: Didn’t we want to get away from ‘LinaroCA’?

Comment (d3zd3z) [reply tools/flow/keys.py]: Yeah, this should probably be something that indicates its "demo" nature.

Also, what was the reason we put the "Signing" part in the OU, but not in the purpose bits of the cert? Or have we just not gotten around to that?

Comment (microbuilder) [reply tools/flow/keys.py]: That sounds like a mistake or oversight.

**Comment by d3zd3z** [comment]
This is a big squash, and now only has two commits with a merge. The merge was done with the git subtree command and would allow us to bring in future updates of the subtree repository later. I've squashed their history, since that isn't really needed for us. It will be important to not let github rebase this when merging it.

**Comment by microbuilder** [comment]
History is much easier to digest this way, thanks for the squash.

File: tools/flow/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Setting things up.
+
+Flow doesn't try to recreate the key creation parts of what Linaro CA
```
Comment (microbuilder) [inline tools/flow/README.md]: LITE Bootstrap Server now (https://github.com/Linaro/lite_bootstrap_server)

Comment (d3zd3z) [reply tools/flow/README.md]: Fixed this.

**Review by microbuilder — APPROVED** [review]

**Pull request merged**